yt/fields/vector_operations.py

```diff
@@ -599,6 +599,66 @@ def _cartesian_z(field, data):
                 display_field=True,
             )
 
+    if registry.ds.geometry == "spherical":
+
+        def _cylindrical_radius_component(field, data):
+            return data[(ftype, f"{basename}_r")] * np.sin(data[(ftype, "theta")])
+
+        registry.add_field(
+            (ftype, f"{basename}_cylindrical_radius"),
+            sampling_type="local",
+            function=_cylindrical_radius_component,
+            units=field_units,
+            display_field=True,
+        )
+
+        registry.alias(
+            (ftype, f"{basename}_cylindrical_z"),
+            (ftype, f"{basename}_cartesian_z"),
+        )
+
+        def _poloidal_magnitude(field, data):
+            return np.sqrt(
+                data[(ftype, f"{basename}_r")] ** 2
+                + data[(ftype, f"{basename}_theta")] ** 2
+            )
+
+        registry.add_field(
+            (ftype, f"{basename}_poloidal_magnitude"),
+            sampling_type="local",
+            function=_poloidal_magnitude,
+            units=field_units,
+            display_field=True,
+        )
```

Comment:
this is not necessary for the rest of the PR, I'm just throwing it in while I'm in there